File: app/src/main/java/com/ivan/github/account/IUserCenter.java
package com.ivan.github.account;

import com.ivan.github.account.model.User;

public interface IUserCenter {

    int STATUS_LOGIN = 1;

    int STSTUS_LOGOUT = 0;

    void registerLoginStatusChangedListener(ILoginStatusChangedListener listener);

    void unregisterLoginStatusChangedListener(ILoginStatusChangedListener listener);

    boolean isLogin();

    User getUser();

    String getUsername();

    String getLogin();

    void logout();

    String getAuthorization();

}

File: app/src/main/kotlin/com/github/account/IUserCenter.kt
package com.github.account

import com.github.account.model.User
import kotlinx.coroutines.flow.Flow

/**
 * User center interface for managing user login status and information.
 */
interface IUserCenter {

    companion object {
        const val STATUS_LOGIN = 1
        const val STATUS_LOGOUT = 0
    }

    /**
     * Check if user is logged in.
     */
    fun isLogin(): Boolean

    /**
     * Reactive login status.
     */
    fun isLoginFlow(): Flow<Boolean>

    /**
     * Get current logged-in user.
     */
    fun getUser(): User?

    /**
     * Get username of current user.
     */
    fun getUsername(): String

    /**
     * Get login name of current user.
     */
    fun getLogin(): String

    /**
     * Get authorization token.
     */
    fun getAuthorization(): String?

    /**
     * Logout current user and clear data.
     */
    fun logout()

    /**
     * Initialize user center with user info and auth token.
     */
    fun init(user: User?, auth: String?)

    /**
     * Save/Update user info.
     */
    fun saveUser(user: User?)

    /**
     * Register login status listener.
     */
    fun registerLoginStatusChangedListener(listener: ILoginStatusChangedListener?)

    /**
     * Unregister login status listener.
     */
    fun unregisterLoginStatusChangedListener(listener: ILoginStatusChangedListener?)
}
